lib/Sema/PreCheckExpr.cpp

```diff
@@ -238,6 +238,32 @@ static bool diagnoseRangeOperatorMisspell(DiagnosticEngine &Diags,
   return false;
 }
 
+static bool diagnoseNonexistentPowerOperator(DiagnosticEngine &Diags,
+                                        UnresolvedDeclRefExpr *UDRE,
```

Comment:
Nit: Formatting here needs to be adjusted, please use clang-format or git-clang-format to fix it.